silq/tools/plot_tools.py

```diff
@@ -807,6 +814,11 @@ def __init__(self,
             subplots = 1
 
         self.point = None
+
+        self.buffer_length = buffer_length
+        self.buffers = [None, ] * self.buffer_length
+        self.buf_idx = 0
```

Comment:
Shouldn't buf_idx be reset in update_plot when initialize=False?

Reply:
it is reset there as well, but I see your point of no need to define it here as well then, can correct.

Reply:
Ah no if it's reset somewhere else then keep it here.
Generally it's good practice to define all class attributes in the __init__

Reply:
ok